jobs/build/ocp/Jenkinsfile

```diff
@@ -642,11 +661,81 @@ Please direct any questsions to the Continuous Delivery team (#aos-cd-team on IR
                 sh "ose_images.sh --user ocp-build build_container --branch rhaos-${BUILD_VERSION}-rhel-7 --group base --repo https://raw.githubusercontent.com/openshift/aos-cd-jobs/master/build-scripts/repo-conf/aos-unsigned-building.repo"
             }
 
-            buildlib.oit """
+            waitUntil {
+              try {
+                exclude = ""
+                if ( BUILD_EXCLUSIONS != "" ) {
+                  exclude = "-x ${BUILD_EXCLUSIONS} --ignore-missing-base"
+                }
+                buildlib.oit """
 --working-dir ${OIT_WORKING} --group openshift-${BUILD_VERSION}
+${exclude}
 images:build
 --push-to-defaults --repo-type unsigned
 """
+                return true // finish waitUntil
+              }
+              catch ( err ) {
+                record_log = buildlib.parse_record_log( OIT_WORKING )
+                builds = record_log['build']
+                failed_map = [:]
+                for(i = 0; i < builds.size(); i++) {
+                  distgit = build[i]['distgit']
+                  if ( builds[i]['status'] == '-1' ){
+                    task_url = "No Brew task available, see jenkins log."
+                    if ( build[i].containsKey('task_url') ) {
+                      task_url = build[i]['task_url']
+                    }
+                    failed_map[distgit] = task_url
+                  }
+                  else {
+                    // build may have succeeded later. If so, remove.
+                    if ( failed_map.containsKey(distgit) ) {
+                      failed_map.remove(distgit)
+                    }
+                  }
+                }
+                
+                failed_list = buildlib.mapToList(failed_map)
```

Comment:
Starting on this line, I'd like to remove as much logic as we can from the pipeline. As we know, loc == pain  when it comes to Jenkins pipelines. To that end, can't we:
- just initialize exclusion_list with failed_map.keys()
- use "${failed_map}"  in the email. Groovy does a sufficient job of printing the map. It will be JSON-y, but this is an internal email and I'm not concerned with it being pretty.

Reply:
Sure, that gets rid of 10 lines.